supabase/migrations/20250422235033_add-cashu-xpubs-to-user.sql

```diff
@@ -0,0 +1,140 @@
+
+alter table "wallet"."users" add column "cashu_locking_xpub" text not null;
+
+CREATE UNIQUE INDEX users_cashu_locking_xpub_key ON wallet.users USING btree (cashu_locking_xpub);
+
+alter table "wallet"."users" add constraint "users_cashu_locking_xpub_key" UNIQUE using index "users_cashu_locking_xpub_key";
+
+-- Drop the old function before creating the new one with the additional parameter
+drop function if exists "wallet"."upsert_user_with_accounts"(p_user_id uuid, p_email text, p_email_verified boolean, p_accounts jsonb[]);
+
+set check_function_bodies = off;
+
+-- update to include cashu_locking_xpub
+CREATE OR REPLACE FUNCTION wallet.upsert_user_with_accounts(p_user_id uuid, p_email text, p_email_verified boolean, p_accounts jsonb[], p_cashu_locking_xpub text)
+ RETURNS jsonb
+ LANGUAGE plpgsql
+AS $function$declare
+  result_user jsonb;
+  existing_accounts jsonb;
+  acct jsonb;
+  new_acc wallet.accounts%ROWTYPE;
+  added_accounts jsonb := '[]'::jsonb;
+  usd_account_id uuid := null;
+  btc_account_id uuid := null;
+begin
+  -- Upsert user
+  insert into wallet.users (id, email, email_verified, cashu_locking_xpub)
+  values (p_user_id, p_email, p_email_verified, p_cashu_locking_xpub)
+  on conflict (id) do update set
+    email = coalesce(EXCLUDED.email, wallet.users.email),
+    email_verified = EXCLUDED.email_verified,
+    cashu_locking_xpub = coalesce(EXCLUDED.cashu_locking_xpub, wallet.users.cashu_locking_xpub);
```

Comment:
I wonder if we should leave the xpub out of the update part? if some changed value would end up here it would override the old one and tokens locked with previous ones would the become unlockable right?

Reply:
but don't we read from here when constructing derinvation path so if this value changes the derivation path would also change right?